### ISSUE TITLE
Error applying plan with aws_appautoscaling_policy : does not have attribute 'arn'

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#404]
_This issue was originally opened by @yesteph as hashicorp/terraform#10486. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hello,

With terraform 0.7.13, we declared an aws_appautoscaling_policy :
_resource "aws_appautoscaling_policy" "scale_in" {
  name = "${aws_ecs_service.mlk_worker_sqs_to_post.name}-scale-in"
  service_namespace = "ecs"
  resource_id = "service/${var.cluster_name}/${aws_ecs_service.mlk_worker_sqs_to_post.name}"
  scalable_dimension = "ecs:service:DesiredCount"

  adjustment_type = "ChangeInCapacity"
  cooldown = 60
  metric_aggregation_type = "Average"

  step_adjustment {
    metric_interval_upper_bound = 0
    scaling_adjustment = -2
  }

  depends_on = ["aws_appautoscaling_target.scale_tgt"]
}_

Sometimes, we got the following error : 

_Error applying plan:

1 error(s) occurred:

* Resource 'aws_appautoscaling_policy.scale_in' does not have attribute 'arn' for variable 'aws_appautoscaling_policy.scale_in.arn'__

Strangely, the policy is correctly created.
A second terraform apply is ok.

**Comment by bflad** [comment]
Hi folks 👋 Sorry for the unexpected behavior here. It turns out there was a very subtle bug that prevented proper recreation of Application AutoScaling Policies when only the `resource_id` argument changed. We also now properly ignore `ObjectNotFoundException` errors during Terraform resource destroy. A full writeup of these changes can be found in #7982. These fixes will be released in version 2.3.0 of the Terraform AWS Provider in the next day or two.

Even without upgrading your Terraform AWS Provider to the newer version, you _may_ be able to workaround the original issue with a simple configuration update. Using direct references to the relevant `aws_appautoscaling_target` resource for the `resource_id`, `scalable_dimension`, and `service_namespace` arguments in the `aws_appautoscaling_policy` resource configuration should be enough in many cases to ensure Terraform has proper ordering information to handle recreations of the Application AutoScaling Policy resource when the underlying Application AutoScaling Target or resource associated with the target changes:

```hcl
resource "aws_appautoscaling_target" "example" {
  # ... other configuration ...
}

resource "aws_appautoscaling_policy" "example" {
 # ... other configuration ...

  resource_id        = "${aws_appautoscaling_target.example.resource_id}"
  scalable_dimension = "${aws_appautoscaling_target.example.scalable_dimension}"
  service_namespace  = "${aws_appautoscaling_target.example.service_namespace}"
}
```

If you are still having trouble after upgrading to version 2.3.0 of the Terraform AWS Provider (when its released) and with a configuration looking similar to the above, please create a new GitHub issue with the relevant details from the issue template and we can further triage. Thanks!

**Comment by bflad** [comment]
This has been released in [version 2.3.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#230-march-21-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
